satpy/scene.py

```diff
@@ -41,6 +41,26 @@
 LOG = logging.getLogger(__name__)
 
 
+def _get_area_resolution(area):
+    """Attempt to retrieve resolution from AreaDefinition."""
+    try:
+        resolution = max(area.pixel_size_x, area.pixel_size_y)
+    except AttributeError:
+        resolution = max(area.lats.resolution, area.lons.resolution)
+    return resolution
+
+
+def _aggregate_data_array(data_array, **coarsen_kwargs):
+    """Aggregate xr.DataArray."""
+    func = coarsen_kwargs.pop("func")
```

Comment:
I think it's better to have the signature like this then: `def _aggregate_data_array(data_array, func, **coarsen_kwargs)`

Reply:
Done